### PR TITLE
Basic Render Profiling

### PR DESCRIPTION
**Pull request by jojolepro**
Adds more profiling to the renderer.

### CODE REVIEW

**Comment by jojolepro** [comment]
Note from omni-viral: There's already a feature gate in the thread profiler macro. We should forward the feature gate to it and remove the feature gate we are using when calling the macro

**Comment by jojolepro** [comment]
https://github.com/glennw/thread_profiler/blob/master/src/lib.rs#L23
line in question for the feat gate

**Comment by torkleyy** [comment]
Postponing until after 0.10.

File: examples/ui/main.rs [hunk 1]
```diff
@@ -67,18 +69,43 @@ impl SimpleState for Example {
...
+            if let Some(random_text) = self.random_text.and_then(|entity| ui_text.get_mut(entity)) {
+                if let Ok(value) = random_text.text.parse::<i32>() {
+                    let mut new_value = value * 10;
+                    if new_value > 100000 {
+                        new_value = 1;
+                    }
+                    random_text.text = new_value.to_string();
+                } else {
+                    random_text.text = String::from("1");
+                }
```
[the remark this answers is not in the document]
Comment (Xaeroxe) [reply examples/ui/main.rs]: Cool, but can we call it something other than random text?

File: examples/ui/resources/display.ron [hunk 2]
```diff
@@ -6,5 +6,5 @@
   multisampling: 0,
   title: "UI example",
   visibility: true,
-  vsync: true,
+  vsync: false,
```
Comment (Xaeroxe) [inline examples/ui/resources/display.ron]: Probably shouldn't leave this

Comment (jojolepro) [reply examples/ui/resources/display.ron]: It was to see the framerate drops caused by gfx-glyph when changing a text dimension. I can put it back to true.

Comment (jojolepro) [reply examples/ui/resources/display.ron]: fixed

**Comment by Moxinilian** [comment]
Is this still currently unfinished?
@Xaeroxe are your concerns resolved?

**Comment by jojolepro** [comment]
Yeah its ready. Its not perfect but I won't improve it more for the time being.

**Comment by Moxinilian** [comment]
bors r-
Thats a lot of commits actually, would you mind squashing them?

**Comment by torkleyy** [comment]
@jojolepro Conflict with upstream :(

**Comment by jojolepro** [comment]
Should be fixed

**Comment by Xaeroxe** [comment]
@jojolepro now we have a problem with the build :(

**Comment by jojolepro** [comment]
Can't see the result, so.. :P

**Comment by fhaynes** [comment]
C'mon TeamCity, why aren't you noticing this one

**Comment by fhaynes** [comment]
Merging this, builds fine now. Think it was a Concourse issue.

**Comment by fhaynes** [comment]
This is failing with a bunch of errors like the following:

```
error: cannot find macro `profile_scope!` in this scope
   --> amethyst_ui/src/pass.rs:114:9
    |
114 |         profile_scope!("ui_pass_build");
    |         ^^^^^^^^^^^^^
```

**Comment by jojolepro** [comment]
Probably the squash changed the Cargo.toml

**Comment by jojolepro** [comment]
no point in forcing it, I need to fix the pr

**Comment by Xaeroxe** [comment]
Pretty sure that will fix it, so I'm going to 

bors r+

if I'm wrong this just won't get merged 🙂

**Pull request merged**